posturebest/ViewModel/BLEManager.swift

```diff
@@ -0,0 +1,116 @@
+//
+//  BLEManager.swift
+//  posturebest
+//
+//  Created by Kaitlyn O’Flaherty on 10/2/24.
+//
+
+import Foundation
+import CoreBluetooth
+
+class BLEManager: NSObject, ObservableObject, CBCentralManagerDelegate, CBPeripheralDelegate {
+    var myCentral: CBCentralManager!
+    @Published var isSwitchedOn = false
+    @Published var periperals = [Peripheral]() // stores discovered periphs
+    @Published var connectedPeripheralUUID: UUID?
+    var cbPeripheral: CBPeripheral?
+    
+    override init() {
+        super.init()
+        myCentral = CBCentralManager(delegate: self, queue: nil)
+    }
+    
+    // Delegate method called when state of central manager updated
+    func centralManagerDidUpdateState(_ central: CBCentralManager) {
+        isSwitchedOn = central.state == .poweredOn
+        if isSwitchedOn {
+            startScanning()
+        } else {
+            stopScanning()
+        }
+    }
+    
+    // Delegate method for when peripheral discovered
+    func centralManager(_ central: CBCentralManager, didDiscover peripheral: CBPeripheral, advertisementData: [String: Any], rssi RSSI: NSNumber) {
+        let newPeripheral = Peripheral(id: peripheral.identifier, name: peripheral.name ?? "Unknown", rssi: RSSI.intValue)
+        
+        // Add peripheral to list if not there
+        if !periperals.contains(where: {$0.id == newPeripheral.id}) {
+            DispatchQueue.main.async{
+                self.periperals.append(newPeripheral)
+            }
+        }
+    }
+    
+    func startScanning() {
+        print("Scanning for devices...")
+        // Right now not scanning for specific services... can be edited later
+        myCentral.scanForPeripherals(withServices: nil, options: nil)
+    }
+    
+    func stopScanning() {
+        print("Scanning stopped.")
+        myCentral.stopScan()
+    }
+    
+    func connect(to peripheral: Peripheral) {
+        guard let cbPeripheralTemp = myCentral.retrievePeripherals(withIdentifiers: [peripheral.id]).first
+            else {
+            print("Peripheral not found for connection")
+            return
+            }
+        
+        // setting the UUID
+        connectedPeripheralUUID = cbPeripheralTemp.identifier
+        cbPeripheralTemp.delegate = self
+
+        cbPeripheral = cbPeripheralTemp
+
+        myCentral.connect(cbPeripheral!, options: nil)
+    }
+    
+    // Delegate method for when peripheral is connected
+    func centralManager(_ central: CBCentralManager, didConnect peripheral: CBPeripheral) {
+        print("Successfully connected to \(peripheral.name ?? "Unknown")")
+        
+        // discover services on the connected peripheral, update later?
```

Comment:
```suggestion
        // discover services on the connected peripheral
```
why would we update later? is this comment outdated?

Reply:
oh yeah I was marking that for when we needed to do specific stuff with the services but i can remove that for now